docs/developer/architecture.rst

```diff
@@ -238,39 +142,134 @@ This abstraction separates the actual data movement and function application fro
-    - The compute kernel/in-memory format is a pandas DataFrame.
-- :doc:`Pandas on Dask </UsingPandasonDask/index>`
+    - The storage format is `pandas` and in-memory partition type is a pandas DataFrame.
+    - More for execution flow see on :doc:`pandas on Ray </flow/modin/core/execution/ray/implementations/pandas_on_ray/index>` page.
+- :doc:`pandas on Dask </UsingPandasonDask/index>`
     - Uses the `Dask Futures`_ execution framework.
-    - The compute kernel/in-memory format is a pandas DataFrame.
-- :doc:`Omnisci </UsingOmnisci/index>`
+    - The storage format is `pandas` and in-memory partition type a pandas DataFrame.
+    - More for execution flow see on :doc:`pandas on Dask </flow/modin/core/execution/dask/implementations/pandas_on_dask/index>` page.
+- pandas on Python
+    - Uses native python execution and serves for debug purposes.
+    - The storage format is `pandas` and in-memory partition type a pandas DataFrame.
+    - More for execution flow see on :doc:`pandas on Python </flow/modin/core/execution/python/implementations/pandas_on_python/index>` page.
+- :doc:`OmniSci on Native </UsingOmnisci/index>` (experimental)
     - Uses OmniSciDB as an engine.
-    - The compute kernel/in-memory format is a pyarrow Table or pandas DataFrame when defaulting to pandas.
+    - The storage format is `omnisci` and in-memory partition type is a pyarrow Table or pandas DataFrame when defaulting to pandas.
+    - More for execution flow see on :doc:`OmniSci on Native </flow/modin/experimental/core/execution/native/implementations/omnisci_on_native/index>` page.
 - :doc:`Pyarrow on Ray </UsingPyarrowonRay/index>` (experimental)
     - Uses the Ray_ execution framework.
-    - The compute kernel/in-memory format is a pyarrow Table.
+    - The storage format is `pyarrow` and in-memory partition type a pyarrow Table.
+    - More for execution flow see on :doc:`Pyarrow on Ray </flow/modin/experimental/core/execution/ray/implementations/pyarrow_on_ray>` page.
+- cuDF on Ray (experimental)
+    - Uses the Ray_ execution framework.
+    - The storage format is `cudf` and in-memory partition type a cuDF DataFrame.
+    - More for execution flow see on :doc:`cuDF on Ray </flow/modin/core/execution/ray/implementations/cudf_on_ray/index>` page.
 
 .. _directory-tree:
 
+DataFrame Partitioning
+----------------------
+
+The Modin DataFrame architecture follows in the footsteps of modern architectures for
+database and high performance matrix systems. We chose a partitioning schema that
+partitions along both columns and rows because it gives Modin flexibility and
+scalability in both the number of columns and the number of rows supported. The
+following figure illustrates this concept.
+
+.. image:: /img/block_partitions_diagram.png
+   :align: center
+
+Currently, each partition's memory format is a `pandas DataFrame`_. In the future, we will
+support additional in-memory formats for the backend, namely `Arrow tables`_.
```

Comment:
aren't these lines slightly outdated (looking at omnisci)?

Reply:
Yes, I would update this as follows
```
Currently, main in-memory format of each partition is a `pandas DataFrame`_. Omnisci(<link to omnisci storage format docs>), PyArrow(<link to pyarrow storage format docs>) and cuDF(link to cudf storage format docs) are also supported as experimental in-memory formats in Modin.
```